### PR TITLE
WIP: set-level()

### PR DESCRIPTION
**Pull request by Kokan**
```
rewrite {
  set-level("${.some_prefix.level}");
};
```
Accepts any template that must be resolved into either the level represented via numeric value or text.

Use this *rewrite* in the cisco parser.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by bazsi** [comment]
It would be great if we could replace the name level() with severity, as that's the ten the standards have been using for a long time now.

Do you think it''d be too much to ask to add aliases to the other cases (eg filter, macros, etc) and then name this as set-severity()?

File: scl/cisco/plugin.conf [hunk 1]
```diff
@@ -80,6 +80,7 @@ block parser cisco-triplet-parser(template() prefix()) {
             };
             rewrite { set("${`prefix`facility}-$1" value('`prefix`facility')); };
         };
+	rewrite { set-level("${`prefix`severity}"); };
     };
 };
```
Comment (bazsi) [inline scl/cisco/plugin.conf]: Ah, i get this now. Does this value also match the value in brackets at the front? Eg the <PRI> header?

**Comment by Kokan** [comment]
"Do you think it''d be too much to ask to add aliases to the other cases (eg filter, macros, etc) and then name this as set-severity()?"
Sure, that is a nice idea. But I'll do an other PR for those changes and rename `set-level` here.

File: lib/rewrite/rewrite-set-level.c [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+  log_msg_make_writable(pmsg, path_options);
+
+  log_template_format(self->level, *pmsg, NULL, LTZ_LOCAL, 0, NULL, result);
+
+  const gint level = _convert_level(result);
+  if (level < 0)
+    {
+      msg_warning("Warning: invalid level to set", evt_tag_str("level", result->str), log_pipe_location_tag(&s->super));
+      goto error;
```
[the remark this answers is not in the document]
Comment (Kokan) [reply lib/rewrite/rewrite-set-level.c]: Sure I'll change it.

**Comment by Kokan** [comment]
I'll reopen once the renaming done properly, but I am not working on this actively. (no make sense to keep it open)

**Comment by smortex** [comment]
@Kokan I would benefit from such a feature to better classify messages I collect through PatternDB.  Do yo mind if I take your code to work further on the topic?  Thanks!

**Comment by Kokan** [comment]
That would be great if you could work on this.

**Pull request closed without merging**